moments/Parsing.py

```diff
@@ -0,0 +1,1119 @@
+"""
+Functions for computing the SFS and related quantities from sequence data. 
+"""
+
+from collections import defaultdict
+import copy
+from datetime import datetime
+import gzip
+import numpy as np
+import os
+import sys
+import time
+import warnings
+
+from . import Spectrum_mod
+
+
+# Raise each unique warning only once
+warnings.simplefilter('once')
+
+
+def parse_vcf(
+    vcf_file,
+    pop_mapping=None,
+    pop_file=None,
+    bed_file=None,
+    interval=None,
+    anc_seq_file=None,
+    allow_low_confidence=False,
+    use_AA=False,
+    filters=None,
+    allow_multiallelic=False,
+    sample_sizes=None,
+    mask_corners=True,
+    ploidy=2,
+    verbose=1,
+    folded=False
+):
+    """
+    Parse the SFS from genotype data stored in a VCF file.
+
+    There are several ways to specify ancestral alleles. By default, ``REF``
+    alleles from the VCF are used to polarize sites. When this is done, the
+    SFS should be folded. If `use_AA` is True, then the ``INFO/AA`` VCF field
```

Comment:
maybe "..., the SFS should be folded, because the reference allele typically does not necessarily correspond to the ancestral allele."